### ISSUE TITLE
Context menu option - strict "(Temporary) Allow requests from host to host"

### ISSUE THREAD

**Issue by wilkowy** [RequestPolicyContinued/requestpolicy#794]
Currently the set of automated rule creation choices are:
- block *.host -> *.host
- block * -> *.host
- allow *.host -> *.host
- allow * -> *.host
- allow *.host -> sub.host
- allow * -> sub.host

IMHO there could be also strict „host” (without „*.”), especially when we use default blocking policy without allowing subdomains. However I'm aware that this would add too many options to this tiny menu (maybe a redesign?) and most already confused users would feel confused even more. This also could lead to a big list of rules for inexperienced users.

Currently this can be done only manually.

**Comment by myrdd** [comment]
I'm adding a screenshot for reference. In this case, `allowSameDomain` is `false`.

![v1.0 menu screenshot](https://cloud.githubusercontent.com/assets/4286829/17246412/2293f0f8-558c-11e6-96c3-e08d5acd6bec.png)

---

> IMHO there could be also strict „host” (without „*.”)

I guess you mean `origin-sub.origin-host.com -> dest-sub.dest-host.com`, no? Indeed, I'd do it like that. In fact, I think it should be the default if `allowSameDomain` is false.

Default-deny mode + `allowSameDomain = false` is equivalent to Host-Level-Strictness (#474). Here's a screenshot of the same site as above, but with RP `v0.5.30`:

![v0.5 menu screenshot](https://cloud.githubusercontent.com/assets/4286829/17246751/5804a37a-558e-11e6-8657-78418810da7f.png)

As the screenshot shows, there are two origin-dest combinations:

1. `origin-sub.origin-host.com -> dest-sub.dest-host.com`
2. `* -> dest-sub.dest-host.com`

This is what I'd show in the menu for host-level strictness.

I'd also like to see a special configuration button in the top-right of the v1.0 menu, which will allow to turn on `*.host.com` rules again.

---

> Currently this can be done only manually.

See also #470 and #601.

**Comment by wilkowy** [comment]
> Indeed, I'd do it like that. In fact, I think it should be the default if allowSameDomain is false.

Correct. So if we are on `www.example.com` than there should/could be `allow/block www.example.com to www.example.com` beside of `a/b *.example.com -> *.example.com`.

However if the strictness is both-sided (origin/destination) then for example when we are on `www.example.com` which uses static images from `img.example.com` the context menu should have both rules - `www.example.com -> www.example.com` and `www.example.com -> img.example.com` (as well for any another sub which could be problematic in current menu design). I've used RP 0.5 ages ago and I'm not quite sure how it was implemented there, but there were more strict rules. Of course „*.” rules are welcome and should coexist with strict rules.

**Comment by myrdd** [comment]
> So if we are on `www.example.com` than there should/could be (...) beside of `a/b *.example.com -> *.example.com`.

Maybe I was not exact enough. I'm planning the menu as follows: Depending on your strictness, you will see one of the following combinations:

* `*.origin.com -> *.dest.com` and `* -> *.dest.com`
* `www.origin.com -> static.dest.com` and `* -> static.dest.com`
* `http://www.origin.com -> https://static.dest.com` and `* -> https://static.dest.com`

So there won't be any `*.example.com -> *.example.com` options in full-host mode, unless you use the option I mentioned:

> I'd also like to see a special configuration button in the top-right of the v1.0 menu, which will allow to turn on *.host.com rules again.

---

> (...) for example when we are on `www.example.com` which uses static images from `img.example.com` the context menu should have both rules - `www.example.com -> www.example.com` and `www.example.com -> img.example.com` (as well for any another sub which could be problematic in current menu design). I've used RP 0.5 ages ago and I'm not quite sure how it was implemented there, but there were more strict rules. Of course „*.” rules are welcome and should coexist with strict rules.

We need to make some design choice. You can see the different behaviors of 0.5 and 1.0 in the screenshots of my previous post. In 0.5 all "full" hosts are listed; in 1.0 only the base domains are listed on the left side.

If (on RP 1.0) a base domain with multiple subdomains is selected, the menu looks like below. In the screenshot, `heise.de` is selected.

![screenshot of the RPC 1.0 menu on www.heise.de with "heise.de" selected](https://cloud.githubusercontent.com/assets/4286829/17491182/d2d3edc0-5da6-11e6-9d8e-5908b67531f5.jpg)

Justin made the design choice to hide information on the left side of the menu. In my opinion that is a good choice. Still, the right side of the menu needs improvement, especially in such cases.
